### ISSUE TITLE
many  empty rows of table_deltas from state history plugin output

### ISSUE THREAD

**Issue by gzliudan** [EOSIO/eos#9386]
When I decodes table_deltas data from state history plugin, I find many empty rows, especially account_metadata table. codes as https://github.com/EOSIO/eos/blob/master/plugins/state_history_plugin/state_history_plugin.cpp#L524:
```cpp
            if (undo.old_values.empty() && undo.new_ids.empty() && undo.removed_values.empty())
               return;
            deltas.push_back({});
            auto& delta = deltas.back();
            delta.name  = name;
            for (auto& old : undo.old_values) {
               auto& row = index.get(old.first);
               if (include_delta(old.second, row))
                  delta.rows.obj.emplace_back(true, pack_row(row));
            }
            for (auto& old : undo.removed_values)
               delta.rows.obj.emplace_back(false, pack_row(old.second));
            for (auto id : undo.new_ids) {
               auto& row = index.get(id);
               delta.rows.obj.emplace_back(true, pack_row(row));
            }
```
When undo.old_values is not empty, undo.removed_values and undo.new_ids is empty, but include_delta(old.second, row) is false, the above codes will not output any row.

**Comment by aclark-b1** [comment]
Hi @gzliudan,

Could you please provide more information about the specific problem this is causing for you?  We want to make sure we understand the impact so we can prioritize this appropriately.

**Comment by gzliudan** [comment]
The small problem is than when I decodes data from state history plugin, sometimes empty vectors are received. I feel strange at this issue, I think I got the reason after debug, so I reminds everyone here.

**Comment by sanaraufx** [comment]
Fixed in 2.1
